staging/src/k8s.io/api/node/v1alpha1/types.go

```diff
@@ -114,3 +114,74 @@ type RuntimeClassList struct {
+type NodeLogViewerOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SinceTime is an RFC3339 timestamp from which to show logs.
+	// +optional
+	SinceTime *metav1.Time `json:"sinceTime,omitempty" protobuf:"bytes,1,opt,name=sinceTime"`
+
+	// UntilTime is an RFC3339 timestamp until which to show logs.
+	// +optional
+	UntilTime *metav1.Time `json:"untilTime,omitempty" protobuf:"bytes,2,opt,name=untilTime"`
+
+	// TailLines if set, return up to this many lines (not more than 100k) from the end of the log.
+	// Support for this is implementation specific and currently only available for Linux and Windows service logs.
+	// +optional
+	TailLines *int64 `json:"tailLines,omitempty" protobuf:"varint,3,opt,name=tailLines"`
+
+	// Path is used to retrieve the contents of a literal file relative to the node's logs directory.
+	// If both a path and a service heuristic are specified, the service heuristic is performed.
+	// +optional
+	Path string `json:"path,omitempty" protobuf:"bytes,4,opt,name=path"`
+
+	// Format is used to display service logs in an alternate format.
+	// Support for this is implementation specific. Currently supported only for service logs on Linux.
+	// Valid values are: short, cat, json, short-unix.
+	// TODO: Make this into an enum type
+	// +optional
+	Format string `json:"format,omitempty" protobuf:"bytes,5,opt,name=format"`
+
+	// Pattern filters log entries by the provided regex pattern. On Linux nodes, this pattern will be read as a PCRE2 regex,
+	// on Windows nodes it will be read as a PowerShell regex.
+	// On Linux, if the pattern is all lowercase, matching is case insensitive. Otherwise, matching is case sensitive.
+	// This can be overridden with the PatternCaseSensitive option, see below.
+	// Support for this is implementation specific and currently only available for Linux and Windows service logs.
+	// +optional
+	Pattern string `json:"pattern,omitempty" protobuf:"bytes,6,opt,name=pattern"`
+
+	// PatternCaseSensitive is used to force case sensitivity or insensitivity the pattern for service logs on Linux nodes.
+	// By default case sensitivity of the search is derived from the Pattern string, see above.
+	// +optional
+	PatternCaseSensitive bool `json:"patternCaseSensitive,omitempty" protobuf:"varint,7,opt,name=patternCaseSensitive"`
+
+	// Boot show messages from a specific boot. Allowed values are [-100, 0]. Passing an invalid boot offset will fail
+	// retrieving logs and return an error.
+	// Support for this is implementation specific. Currently supported only for service logs on Linux.
+	// +optional
+	Boot *int64 `json:"boot,omitempty" protobuf:"varint,8,opt,name=boot"`
+
+	// Services are the specified service(s) to return log entries from.
+	// The server implements a heuristic to get the logs from a service or a file.
+	// +optional
+	// +listType=set
+	Services []string `json:"services,omitempty" protobuf:"bytes,9,rep,name=services"`
```

Comment:
What happens if I specify neither Services nor Path?  Is it an error?

Reply:
I took your suggestion of defaulting to returning all services when neither Services or Path is specified.